### ISSUE TITLE
Question: Clarify needs of PackageDependency

### ISSUE THREAD

**Issue by Tirraon** [microsoft/microsoft-ui-xaml#1406]
Hi!
[Getting started](https://docs.microsoft.com/ru-ru/uwp/toolkits/winui/getting-started) page says, that we only need to install NuGet package and setup XamlControlsResources to use WinUI in our applications.
As I understand, whole WinUI package isn't embedded in package (nor for store submission, nor for sideloading). And it ships to user as separated dependency, which will be automatically downloaded from store with application, and should be added to appxinstaller in case of sideloading.
But I found, that we can add explicit dependency in *.appxpackage file, like some devs have to add in following issues: https://github.com/microsoft/microsoft-ui-xaml/issues/999#issuecomment-509784905 https://github.com/microsoft/microsoft-ui-xaml/issues/616#issuecomment-496664803 https://github.com/microsoft/microsoft-ui-xaml/issues/634#issuecomment-488103175.

So, my questions about it:
1. For what reasons I need to add explicit dependency to *.appxpackage file? 
2. If we need to add it, so with which minversion it should be setup (like 'MinVersion="2.11906.6001.0"') ? Looks like it is not related to NuGet package version.
3. Am I right, that store should download correct WinUI dependency without additional setup (comparing with Getting Started page)?
4. Am I right, that visual studio should to create appxinstaller file with WinUI dependencies for every platform (x86, x64, arm, arm64)?
5. What I should also know about it, if I want to use WinUI controls (like NavigationView) on Windows Phone (currently I use 2.1 version to support 10.0.14371)?
6. Is there WinUI docs with information about explicit PackageDependency?

If answers depends on it:
I plan to use WinUI with C# application that targets 1903 and has min target version 1607.

**Comment by jevansaks** [comment]
> 1. For what reasons I need to add explicit dependency to *.appxpackage file?

All these scenarios should be bugs from my perspective. It should "just work". What scenario do you have where you're finding that you need to add it explicitly?



> 3\. Am I right, that store should download correct WinUI dependency without additional setup (comparing with Getting Started page)?

Correct.


> 4\. Am I right, that visual studio should to create appxinstaller file with WinUI dependencies for every platform (x86, x64, arm, arm64)?

It should. Please let us know if you have a scenario where that's not the case and we can follow up with VS.

> 5\. What I should also know about it, if I want to use WinUI controls (like NavigationView) on Windows Phone (currently I use 2.1 version to support 10.0.14371)?

Windows Phone is end-of-life in December and store updates have already been stopped. This isn't supported anymore.

**Comment by ad1Dima** [comment]
> Windows Phone is end-of-life in December and store updates have already been stopped. This isn't supported anymore.
Windows 10 mobile still receive update from store. Developers still can update there apps.

>Window 10 Mobile version 1703 was the last supported OS version and will reach end of support on June 11th, 2019.

**Comment by ad1Dima** [comment]
> What I should also know about it, if I want to use WinUI controls (like NavigationView) on Windows Phone (currently I use 2.1 version to support 10.0.14371)?

My app works great on lumia 950 with WinUI 2.1 and 1607 as minversion

**Comment by jevansaks** [comment]
If WinUI 2.1 works for you, you should feel free to continue to use it. Going forward our minversion will be higher than the max version available on Windows Mobile. Sorry.

Please let me know if you have scenarios you encounter where you find you need to manually add the package dependency. It shouldn't be necessary.

**Comment by maxkatz6** [comment]
@jevanks hello!
Sorry for commenting on closed issue, but I have finally started to use WinUI in my project, and have now one issue.
I build with 1607 as min target and WinUI 2.1. Everything works fine for me and most of my users on all supported devices including Mobile.
But I see in AppCenter crashes from small percent of users with next message and stacktrace:
```
System.Exception: Cannot create instance of type 'Microsoft.UI.Xaml.Controls.XamlControlsResources'

System.Runtime.InteropServices
McgMarshal.ThrowOnExternalCallFailed (Int32 hr, RuntimeTypeHandle typeHnd) f:\dd\ndp\fxcore\CoreRT\src\System.Private.Interop\src\Shared\McgMarshal.cs at 1267:13
__Interop
ComCallHelpers.Call ($__ComObject __this, RuntimeTypeHandle __typeHnd, Int32 __targetIndex) C:\Windows\ServiceProfiles\NetworkService\AppData\Local\Packages\appcompile-20200508-084433-18692\AC\Temp\Ilc1380765118\FSClient.McgInterop\SharedStubs.g.cs at 11369:5
__Interop.ForwardComStubs
Stub_10<System.__Canon> (Void* InstParam, $__ComObject __this, Int32 __targetIndex) C:\Windows\ServiceProfiles\NetworkService\AppData\Local\Packages\appcompile-20200508-084433-18692\AC\Temp\Ilc1380765118\FSClient.McgInterop\SharedStubs.g.cs at 392:4
Microsoft.AppCenter.Utils
ApplicationLifecycleHelper.<ctor>b__17_1 (Object sender, $UnhandledErrorDetectedEventArgs eventArgs) at 0
```

I have read related issues. And found problems with VS 2017 (but I build it with 2019) and missing PackageDependency in appx manifest (but I have checked appxupload file before sending to store - it has correct dependencies for every platform).
Could it be problem with Net Native 1.7, which is used for 1607 min target? 
Could it be resolved with latest WinUI 2.4 and 1709 min target? I think, I can maintain two packages with different min targets for that.

**Comment by maxkatz6** [comment]
Probably it is related to https://github.com/microsoft/microsoft-ui-xaml/issues/1712. As I understand there is a bug in Windows, which is fixed with current insider (May 2020 Update?). But will it be fixed for older Windows versions?
